### PR TITLE
Update to ESLint 9.11 and typescript-eslint 8.11

### PR DESCRIPTION
**Pull request by wgoehrig**
Since ESLint v8.x reaches [end of life ***this weekend***](https://eslint.org/blog/2024/09/eslint-v8-eol-version-support/), it's time for us to start work on updating to v9.

This is currently **blocked** waiting on two dependencies to update:
- [eslint-plugin-import](https://github.com/import-js/eslint-plugin-import/issues/2948) - PR is merged, release expected in the next few days.
- [eslint-plugin-react-hooks](https://github.com/facebook/react/issues/28313) - Also merged, but only "officially" releasing with React 19, which has been delayed for months.  We may have to use the beta, or opt for the [ESLint compatibility utilities](https://eslint.org/blog/2024/05/eslint-compatibility-utilities/).

So far I've only tested these changes in itwinjs-core.

We will also need to decide what to do about stylistic rules - these have moved out of typescript-eslint to [@stylistic/eslint-plugin-ts](https://eslint.style/), but the stylistic rules overall are [discouraged by typescript-eslint](https://typescript-eslint.io/users/what-about-formatting).  We may want to consider prettier or [dprint](https://dprint.dev/)...

### CODE REVIEW

File: package.json
Comment (aruniverse) [inline package.json:42]: wonder if we should remove this, hasnt been touched in 5 yrs

File: dist/rules/import-spacing.js
Comment (aruniverse) [inline dist/rules/import-spacing.js:1]: does this eslint rule seem silly?
should this be handled by prettier or some other formatter?

File: dist/configs/itwinjs-recommended.js
Comment (aruniverse) [inline dist/configs/itwinjs-recommended.js:190]: i would lke to vote for no, and defer to using a formatter like prettier or biome or something else

Comment (wgoehrig) [reply dist/configs/itwinjs-recommended.js]: agreed.  I went ahead and removed the other deprecated stylistic rules as well

File: package.json
Comment (wgoehrig) [reply package.json:42]: we have an issue open for that (https://github.com/iTwin/eslint-plugin/issues/74), but I think we should consider that out of scope for this PR

File: dist/rules/import-spacing.js
Comment (wgoehrig) [reply dist/rules/import-spacing.js:1]: It's definitely silly if we're gonna adopt a formatter!  Let's remove it

File: dist/configs/itwinjs-recommended.js
Comment (aruniverse) [inline dist/configs/itwinjs-recommended.js:196]:
```suggestion
```

**Review by aruniverse — COMMENTED** [review]
lgtm, but missing beachball changes
i assume this should be a new major ver of this pkg, and wed want to start off with pre-releases?

**Comment by wgoehrig** [comment]
> lgtm, but missing beachball changes i assume this should be a new major ver of this pkg, and wed want to start off with pre-releases?

Just added the beachball changes, but might need some help w/ publishing pre-releases if you all think those are necessary...

**Comment by ben-polinsky** [comment]
> > lgtm, but missing beachball changes i assume this should be a new major ver of this pkg, and wed want to start off with pre-releases?
> 
> Just added the beachball changes, but might need some help w/ publishing pre-releases if you all think those are necessary...

Just a GH action to trigger, I can do it after merge.

**Review by aruniverse — APPROVED** [review]

**Pull request merged**
